src/app/auth.guard.ts

```diff
@@ -3,18 +3,24 @@ import { Injectable } from "@angular/core";
 import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot, UrlTree } from "@angular/router";
 
 import { AuthenticationService } from "./shared/services/authentication.service";
+import { Observable, map } from "rxjs";
 
 
 @Injectable()
 export class AuthGuard implements CanActivate {
-  constructor(private authService: AuthenticationService, private router: Router) {}
+  constructor(private authService: AuthenticationService, private router: Router){}
 
-  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
-    if (this.authService.isAuthenticated()) {
-      return true;
-    } else {
-      this.router.navigate(['/login']);
-      return false;
-    }
+  canActivate(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): 
+    Observable<boolean> | Promise<boolean> | boolean  {
+    console.log('auth guard');
+    return this.authService.isAuthenticated$.pipe(
+      map(isAuth => {
+        if (isAuth) {
+          return true;
+        } else {
+          return false;
```

Comment:
Suppose it can be simplified to `return !!isAuth`

Reply:
I left is it was, because I add the routing from AuthGuard to LoginPage if `Auth == false`